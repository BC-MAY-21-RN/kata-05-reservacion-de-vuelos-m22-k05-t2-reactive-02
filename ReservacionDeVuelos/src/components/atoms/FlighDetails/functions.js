const splitDate = date => {
  var arrayDate = date.split('-');
  return monthFilter(arrayDate[1]) + ' ' + arrayDate[2] + ', ' + arrayDate[0];
};

const monthFilter = month => {
  switch (month) {
    case '01':
      return 'January';
    case '02':
      return 'february';
    case '03':
      return 'March';
    case '04':
      return 'April';
    case '05':
      return 'May';
    case '06':
      return 'June';
    case '07':
      return 'July';
    case '08':
      return 'August';
    case '09':
      return 'September';
    case '10':
      return 'October';
    case '11':
      return 'November';
    case '12':
      return 'December';
  }
};

const functions = {splitDate};

export default functions;
