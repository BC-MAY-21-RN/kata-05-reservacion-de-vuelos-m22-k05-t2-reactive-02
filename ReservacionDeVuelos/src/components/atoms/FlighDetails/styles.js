import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    borderBottomColor: 'gray',
    marginHorizontal: 30,
    paddingBottom: 20,
    borderBottomWidth: 1,
    borderColor: 'black',
  },
  details: {
    fontSize: 15,
    fontWeight: 'bold',
  },
});

export default styles; 