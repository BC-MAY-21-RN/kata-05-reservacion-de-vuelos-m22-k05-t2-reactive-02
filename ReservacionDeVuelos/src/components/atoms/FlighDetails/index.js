import React from 'react';
import {View, Text} from 'react-native';
import styles from './styles';
import functions from './functions';

const FlightsDetails = ({data}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.details}>{functions.splitDate(data.fecha)}</Text>
      <Text style={styles.details}>
        {data.passengers} {data.passengers === undefined ? '' : 'passangers'}
      </Text>
    </View>
  );
};

export default FlightsDetails;
