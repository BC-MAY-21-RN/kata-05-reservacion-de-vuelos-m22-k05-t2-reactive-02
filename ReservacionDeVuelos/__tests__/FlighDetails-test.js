import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import FlighDetails from '../src/components/atoms/FlighDetails';

describe('Tests FlighDetails component', () => {
  it('FlighDetails snapshot', () => {
    const renderFlighDetails = renderer
      .create(<FlighDetails data={{passengers: 4, fecha: '2022-05-20'}} />)
      .toJSON();
    expect(renderFlighDetails).toMatchSnapshot();
  });
});
