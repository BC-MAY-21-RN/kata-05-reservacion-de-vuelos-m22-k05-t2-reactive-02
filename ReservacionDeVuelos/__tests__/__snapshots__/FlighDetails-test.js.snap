// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests FlighDetails component FlighDetails snapshot 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "gray",
      "borderBottomWidth": 1,
      "borderColor": "black",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginHorizontal": 30,
      "paddingBottom": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "bold",
      }
    }
  >
    May 20, 2022
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "bold",
      }
    }
  >
    4
     
    passangers
  </Text>
</View>
`;
